profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.multilingual.inc

```diff
@@ -0,0 +1,701 @@
+  $config['FeedsHTTPFetcher']['source'] = NewsroomHelper::restoreFeedImporter($importer->id);
+  $source->setConfig($config);
+  $source->save();
+}
+
+/**
+ * Helper prepares the tamper settings.
+ *
+ * @param string $settings
+ *   Tamper settings.
+ * @param int $i
+ *   The order in the parser.
+ * @param string $type
+ *   Taxonomy or node.
+ *
+ * @return array
+ *   Tamper settings.
+ */
+function _nexteuropa_newsroom_prepare_dynamic_tamper($settings, $i, $type = NULL) {
+  $str = 'xpathparser_' . $i;
+  $tamper = array(
+    'id' => $type . '-' . $str . '-' . $settings['plugin_id'],
+    'importer' => $type,
+    'source' => 'xpathparser:' . $i,
+    'plugin_id' => $settings['plugin_id'],
+  );
+
+  // Prepare setting for Entity Field Query finder.
+  if ($settings['plugin_id'] == 'efq_finder') {
+    $tamper += array(
+      'settings' => array(
+        'entity_type' => 'taxonomy_term',
+        'bundle' => $settings['vocabulary'],
+        'column' => 'value',
+        'property' => FALSE,
+        'update' => 'Update',
+        'field' => $settings['field'],
+      ),
+      'description' => 'Entity Field Query finder',
+    );
+  }
+
+  // Prepare setting for Rewrite plugin.
+  if ($settings['plugin_id'] == 'rewrite') {
+    $tamper += array(
+      'settings' => array(
+        'text' => $settings['prefix'] . '[xpathparser:' . $i . ']',
+      ),
+      'description' => 'Rewrite',
+    );
+  }
+
+  // String to timestamp plugin.
+  if ($settings['plugin_id'] == 'strtotime') {
+    $tamper += array(
+      'settings' => '',
+      'description' => 'String to Unix timestamp',
+    );
+  }
+
+  return $tamper;
+}
+
+/**
+ * Build tampers programmatically.
+ */
+function _nexteuropa_newsroom_dynamic_tamper($tamper) {
+  $def_lang = language_default()->language;
+  $feeds_tamper = new stdClass();
+  $feeds_tamper->disabled = FALSE;
+  $feeds_tamper->api_version = 2;
+  $feeds_tamper->weight = 0;
+  if ($tamper == NULL) {
+    $feeds_tamper->id = 'newsroom_items_multilingual-blank_source_1-default_value';
+    $feeds_tamper->importer = NEXTEUROPA_NEWSROOM_ITEM_IMPORTER;
+    $feeds_tamper->source = 'Blank source 1';
+    $feeds_tamper->plugin_id = 'default_value';
+    $feeds_tamper->settings = array(
+      'default_value' => $def_lang,
+    );
+    $feeds_tamper->description = 'Set default language';
+  }
+  else {
+    $feeds_tamper->id = $tamper['id'];
+    $feeds_tamper->importer = $tamper['importer'];
+    $feeds_tamper->source = $tamper['source'];
+    $feeds_tamper->plugin_id = $tamper['plugin_id'];
+    $feeds_tamper->settings = $tamper['settings'];
+    $feeds_tamper->description = $tamper['description'];
+  }
+
+  // Delete a pre-existing tamper with the same id.
+  db_delete('feeds_tamper')
+    ->condition('id', $feeds_tamper->id)
+    ->execute();
+  // Save the tamper.
+  feeds_tamper_save_instance($feeds_tamper);
+}
+
+/**
+ * Helper function to get the enabled languages.
+ */
+function _nexteuropa_newsroom_languages() {
```

Comment:
@tatarbj I added static cache, because we call this code several times in one request. I don't think, that we should save it as static variable in this case. This code called only when we create feed importers and other case when we add new fields to importers. In other cases this code never called